include/circt/Dialect/HW/HWAttributesNaming.td

```diff
@@ -10,25 +10,50 @@
 //
 //===----------------------------------------------------------------------===//
 
-def InnerRefAttr : AttrDef<HWDialect, "InnerRef"> {
+def InnerRefAttr : AttrDef<HWDialect, "InnerRef", [
+    DeclareAttrInterfaceMethods<SubElementAttrInterface>]> {
   let summary = "Refer to a name inside a module";
   let description = [{
     This works like a symbol reference, but to a name inside a module.
   }];
   let mnemonic = "innerNameRef";
-  let parameters = (ins "::mlir::StringAttr":$module, "::mlir::StringAttr":$name);
+  let parameters = (ins "::mlir::SymbolRefAttr":$symbol);
```

Comment:
module should certainly be a FlatSymbolReference, but the name inside cannot be.  This is because it isn't a symbol, as it's not an immediate child of a symbol table op.  Modules cannot be symbol tables as then instances would be unable to resolve module symbols in the parent scope.  Packing both the module and op into a symbol reference is both wrong and misleading.

That said, module can have the type updated and the attrinterface can be added here.

Reply:
Good point. It's too bad that MLIR symbol tables not what we need -- it's bitten me many times.

Reply:
It's still useful if we want to track the uses of a module.